D/D.sublime-syntax

```diff
@@ -8,339 +8,2031 @@ file_extensions:
+    - include: not-whitespace-illegal
+  version-condition-end:
+    - match: '\)'
+      scope: punctuation.section.brackets.end.d
+      pop: true
+    - include: not-whitespace-illegal
+
+  debug-condition:
+    - match: '\('
+      scope: punctuation.section.brackets.begin.d
+      set:
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: version-condition-end
+        - match: '{{integer_lookahead}}'
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - match: '(?=\S)'
+      pop: true
+
+  static-foreach-in:
+    - match: '\b(static)\s+(foreach|foreach_reverse)\b'
       captures:
-        1: punctuation.definition.comment.d
-  constant_placeholder:
-    - match: '(?i:%(\([a-z_]+\))?#?0?\-?[ ]?\+?([0-9]*|\*)(\.([0-9]*|\*))?[hL]?[a-z%])'
-      scope: constant.other.placeholder.d
-  function_assertions:
-    - match: '(in|out)\s*(?=\{)'
+        1: keyword.control.flow.d
+        2: keyword.control.flow.d
+      push: [optional-block-statement, foreach-arguments]
+
+  static-assert-in:
+    - match: '\b(static)\s+(assert)\b'
+      captures:
+        1: keyword.control.conditional.d
+        2: keyword.control.conditional.d
+      push: [expect-end-of-line, assert-arguments]
+
+  constructions-in:
+    - match: '{{construction_lookahead}}'
+      push: construction
+  construction:
+    # Post-blit constructor
+    - match: '\b(this)\s*(\()\s*(this)\s*(\))'
+      captures:
+        1: entity.name.function.post-blit.d
+        2: punctuation.section.group.begin.d
+        3: variable.language.d
+        4: punctuation.section.group.end.d
+      set: function-definition-after-arguments
+    - match: '\b(this)\s*(?=\(|$)'
+      captures:
+        1: entity.name.function.constructor.d
+        2: punctuation.section.group.begin.d
+      set: constructor-argument-list
+    - match: '(~\s*this)\s*(\()\s*(\))'
       captures:
-        1: keyword.control.statement.d
+        1: entity.name.function.destructor.d
+        2: punctuation.section.group.begin.d
+        3: punctuation.section.group.end.d
+      set: function-definition-after-arguments
+    - include: not-whitespace-illegal-pop
+  # This could either be function/template parameters or arguments. Assume
+  # arguments and switch to parameters when possible
+  constructor-argument-list:
+    - match: '\('
+      scope: punctuation.section.group.begin.d
+      set:
+        - match: '\)'
+          scope: punctuation.section.group.end.d
+          set: function-definition-after-first-argument-list
+        - match: '{{parameter_attribute_lookahead}}'
+          set: [function-definition-after-first-argument-list, function-argument-definition-list-content]
+        - match: '(?=\S)'
+          set: [constructor-argument-list-after-value, value]
+    - include: not-whitespace-illegal-pop
+  constructor-argument-list-after-value:
+    - match: ','
+      scope: punctuation.
```

Comment:
Already asked the question in #1228 whether it might make sense to scope `switch` as `keyword.control.conditional` because of two reasons:

1. `keyword.control.flow` is used for many keywords like `goto`, `return`, etc. If they are scoped differently to normal conditional or loop statements, switch would do so as well, even though it might be more related with keywords like if, else, ...
2. A switch-case statement is somehow an efficient replacement of an if-elseif-elseif...else statement.

What do you think about it? 

Most syntaxes use `keyword.control.flow` so far. Hence your choice is very perfect here. I just wondered if it might make sense to change in general (other syntaxes as well).

Reply:
At least in D switch is a lot more complicated than a simple if-elseif-else statement. For instance the following is perfectly valid:

```d
switch (i) {
    case 1:
    {
        case 2:
    }
    break;
}
```

That and you can use `goto case` and `goto default`. The way I see it `switch` is just as much a conditional as `while`. You can even use switch as a loop:

```d
void main() {
    int a;
    switch (1) {
        case 1:
            writeln(a);
            if (a++ == 10)
                break;
            goto case 1;
        default:
            break;
    }
}
```

Reply:
In that case the general `flow` is very perfect.